presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaSplitManager.java

```diff
@@ -100,10 +101,18 @@ public ConnectorPartitionResult getPartitions(ConnectorTableHandle tableHandle,
                         log.warn("No leader for partition %s/%s found!", metadata.topic(), part.partitionId());
                     }
                     else {
+                        // make sure the first broker is the partition leader
+                        List<HostAddress> isrList = part.isr().stream().map(KafkaSplitManager::brokerToHostAddress).collect(toList());
+                        HostAddress partitionLeader = HostAddress.fromParts(leader.host(), leader.port());
+                        if (!isrList.get(0).getHostText().equals(leader.host())
```

Comment:
my concern around all of this is that this may or may not work depending on hostname / ip address etc. Doing a string compare assumes that the `leader.host()` is the same string as what gets returned by `brokerToHostAddress()`.